workflow-examples/src/test/java/com/redhat/parodos/examples/move2kube/task/Move2KubeTaskTest.java

```diff
@@ -0,0 +1,169 @@
+package com.redhat.parodos.examples.move2kube.task;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+import com.redhat.parodos.workflow.utils.WorkContextUtils;
+import com.redhat.parodos.workflows.work.WorkContext;
+import com.redhat.parodos.workflows.work.WorkReport;
+import com.redhat.parodos.workflows.work.WorkStatus;
+import dev.parodos.move2kube.ApiException;
+import dev.parodos.move2kube.api.ProjectInputsApi;
+import dev.parodos.move2kube.api.ProjectsApi;
+import dev.parodos.move2kube.api.WorkspacesApi;
+import dev.parodos.move2kube.client.model.CreateProject201Response;
+import dev.parodos.move2kube.client.model.ProjectInputsValue;
+import dev.parodos.move2kube.client.model.Workspace;
+import lombok.extern.slf4j.Slf4j;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+@Slf4j
+public class Move2KubeTaskTest {
+
+	Move2KubeTask task;
+
+	private WorkspacesApi workspacesApi;
+
+	private ProjectsApi projectsApi;
+
+	private ProjectInputsApi projectInputsApi;
+
+	private static String move2KubeWorkspaceIDCtxKey = "move2KubeWorkspaceID";
+
+	private static String move2KubeProjectIDCtxKey = "move2KubeProjectID";
+
+	@Before
+	public void BeforeEach() {
+		workspacesApi = Mockito.mock(WorkspacesApi.class);
+		projectsApi = Mockito.mock(ProjectsApi.class);
+		projectInputsApi = Mockito.mock(ProjectInputsApi.class);
+
+		task = new Move2KubeTask("http://localhost", workspacesApi, projectsApi, projectInputsApi);
+		log.error("Move2KubeTask BeforeEach");
+	}
+
+	@After
+	public void AfterEach() {
+		log.error("Move2KubeTask AfterEach");
+	}
+
+	@Test
+	public void testValidExecution() {
+		// given
+		WorkContext workContext = getSampleWorkContext();
+		Workspace workspace = getSampleWorkspace("test");
+
+		Map<String, ProjectInputsValue> inputs = new HashMap<String, ProjectInputsValue>();
+		inputs.put("test1", getSampleProjectInput("test1"));
+		inputs.put("test2", getSampleProjectInput("test2"));
+		workspace.setInputs(inputs);
+
+		assertDoesNotThrow(() -> {
+			Mockito.when(workspacesApi.getWorkspaces()).thenReturn(List.of(workspace));
+			Mockito.when(projectsApi.createProject(Mockito.any(), Mockito.any())).thenReturn(getSampleProject("test"));
+		});
+
+		// when
+		WorkReport report = task.execute(workContext);
+
+		// then
+		assertNull(report.getError());
```

Comment:
1. pls prefer the use of assertThat.
2. The order of parameters counts as it used to generate the error message: `assertXYX(expected, actual)`